programs/quarter_board/main.c

```diff
@@ -0,0 +1,199 @@
+#include "ch.h"
+#include "hal.h"
+
+#include <icu_lld.h>
+#include "chprintf.h"
+#include <inttypes.h>
+
+
+#include "communication.h"
+
+#define BIG_BOOLSHIT 3000
+#define SMALL_BOOLSHIT 400
+#define NOT_SEEING_VALUE 3000
+#define QUANTITY_OF_SENSORS 2
+
+
+int32_t send_values[QUANTITY_OF_SENSORS];
+int32_t sensors_values[QUANTITY_OF_SENSORS];
+bool values_available[QUANTITY_OF_SENSORS];
+
+thread_t *commanderp;
+thread_t *ir_measurementp;
+
+/*
+ * ICU callbacks
+ */
+
+static void read_pulse(ICUDriver *icup) {
+    uint8_t i;
+    int32_t value;
+
+    if(icup == &ICUD1) {
+        i = 0;
+    } else {
+        i = 1;
+    }
+    
+    value = icuGetWidthX(icup);
+
+    if(value < BIG_BOOLSHIT && value > SMALL_BOOLSHIT) {    
+        sensors_values[i] = value;
+        values_available[i] = true;
+    } else {
+        values_available[i] = false;
+    }
+}
+
+THD_WORKING_AREA(waThread1, 64);
+THD_FUNCTION(Thread1, arg) {
+    (void)arg;

```

Comment:
I would make a macro for this, so it is more meaningful (and easier to change for different compiler), in this way:
```c
#define UNUSED(x) (void)(x)
...
UNUSED(arg);
```